support/upsert/upsert.go

```diff
@@ -121,10 +159,42 @@ func (p *createOrUpdateProvider) CreateOrUpdate(ctx context.Context, c crclient.
 	if err := c.Update(ctx, obj); err != nil {
 		return controllerutil.OperationResultNone, err
 	}
-	if hasStatusSubResource(obj) {
-		if err := mutate(f, key, obj); err != nil {
+
+	return controllerutil.OperationResultUpdated, nil
+}
+
+// CreateOrUpdateV2 is a copy of CreateOrUpdate with
+// an important difference: It doesn't take a mutate function, and it doesn't override the passed in object
```

Comment:
// an important difference: It doesn't take a mutate function, and it doesn't override the passed in object

Isn't this what the update in l190 does?

Reply:
are you referring to the `defaultDeploymentSpec()`, `defaultServiceSpec()`, etc functions called in the update? 
right, they do override some parts of the object set defaults but they don't completely replace the object with what is stored in the API server as the current `createOrUpdate()` function does.

according to the comment in the code
```
It copies a number of fields from the object
on the server to the mutated object if unset in the latter. This
avoids unnecessary updates when our code sets a whole struct that
has fields that get defaulted by the server.
```

Reply:
sorry I was referring to something else but I see now, I'm trying to think of a better name than v2